src/qcodes/instrument_drivers/rohde_schwarz/SGS100A.py

```diff
@@ -139,7 +138,90 @@ def __init__(self, name: str, address: str, **kwargs: Any) -> None:
                            set_cmd='SOUR:IQ:IMP:QUAD {:.2f}',
                            get_parser=float,
                            vals=vals.Numbers(-8, 8))
-
+        # Determines the signal at the input/output of the multi purpose [TRIG] connector.
+        self.add_parameter('trigger_connector_mode',
+                           label='Trigger Connector Mode',
+                           get_cmd='CONN:TRIG:OMOD?',
+                           set_cmd='CONN:TRIG:OMOD {}',
+                           vals=vals.Enum('SVAL',   # SVALid - Signal valid
+                                          'SNVAL',  # SNValid - Signal not valid
+                                          'PVO',    # PVOut - Pulse video out (K22 Only)
+                                          'PET',    # PETrigger - Pulse mod ext trigger - PETrigger (K22 Only)
+                                          'PEMS',   # PEMSource - Pulse mode ext source (K22 Only)
+                                          'sval', 'snval', 'pvo', 'pet', 'pems'))
```

Comment:
Is the reason to have lower case here to allow the user to both input lower and upper case? If so I think it would be cleaner to add a `set_parser=lambda x: x.upper` to the parameter. That way the user can alway use the lower case but there is only one format internally

Reply:
i didn't know that would work with vals.Enum. also i wanted to be able to roundtrip correctly. i don't have access to the isntrument at this point

Reply:
the instrument accepts both lowercase and uppercase and I didn't want to impose either on the user. set_parser is applied after the vals.Enum so it's not good for what I wanted.

Reply:
it was also unclear whether the instrument would always return the uppercase version but yes i could add a get_parser to ensure that? i dont' know

Reply:
the set_parser is applied after the validator so it won't work for what i wanted; also i don't want all crazy combinations of lowercase and uppercase to work